query-engine/js-connectors/js/js-connector-utils/src/types.ts

```diff
@@ -3,9 +3,28 @@ import { ColumnTypeEnum } from './const'
 export type ColumnType = typeof ColumnTypeEnum[keyof typeof ColumnTypeEnum]
 
 export interface ResultSet {
+  /**
+   * List of column types appearing in a database query, in the same order as `columnNames`.
+   * They are used within the Query Engine to convert values from JS to Quaint values.
+   */
   columnTypes: Array<ColumnType>
+
+  /**
+   * List of column names appearing in a database query, in the same order as `columnTypes`.
+   */
   columnNames: Array<string>
+
+  /**
+   * List of rows retrieved from a database query.
+   * Each row is a list of values, whose length matches `columnNames` and `columnTypes`.
+   */
   rows: Array<Array<any>>
+
+  /**
+   * The last ID of an `INSERT` statement, if any.
+   * This is required for `AUTO_INCREMENT` columns in MySQL and SQLite-flavoured databases.
+   */
+  lastInsertId?: string
 }
```

Comment:
To be fair, the actual type should be:

```typescript
export type Flavour = 'mysql' | 'postgres' | 'sqlite'

export type ResultSet<flavour extends Flavour> = {
  columnTypes: Array<ColumnType>
  columnNames: Array<string>
  rows: Array<Array<any>>
} & (
  flavour extends 'mysql' | 'sqlite'
    ? { lastInsertId: string }
    : {}
)
```

but this would result in plenty of template specialisations in `Connector<>` and `ResultSet<>` that could clutter the codebase a bit. I'm willing to add them only if deemed necessary by the reviewer.

Reply:
I think this is fine for now, `quaint::ResultSet` has identical type.